app/Models/DiscordMessageThreadMapping.php

```diff
@@ -68,30 +71,31 @@ public static function storeMapping(int $messageThreadId, string $discordThreadI
     /**
      * Find an existing Discord thread mapping for a reportable item.
      */
-    public static function findReportMapping(DiscordReportableType $reportableType, int $reportableId): ?self
+    public static function findReportMapping(ModerationReportableType $reportableType, int $reportableId): ?self
     {
-        return self::where('reportable_type', $reportableType->value)
+        $report = UserModerationReport::where('reportable_type', $reportableType->value)
             ->where('reportable_id', $reportableId)
+            ->whereHas('discordThreadMapping')
             ->first();
+
+        return $report?->discordThreadMapping;
     }
 
     /**
-     * Store a new Discord thread mapping for a reportable item.
+     * Store a new Discord thread mapping for a moderation report.
      */
     public static function storeReportMapping(
-        DiscordReportableType $reportableType,
-        int $reportableId,
```

Comment:
`moderationReportId` is going to be unique, so this should never actually do an update.

Was the old implementation trying to maintain a single discord thread regardless of how many unique users reported it?

Reply:
Yes, we should indeed maintain [a single Discord thread per reportable item](https://discord.com/channels/476211979464343552/1441189751364522139).

However, in latest this method is no longer needed and has been removed.